benchmarks/polys.py

```diff
@@ -1,17 +1,123 @@
-import sympy
+from sympy import symbols, prod
+from sympy.polys import ZZ, Poly, ring
+
 
 class TimePolyManyGens:
     """Time using a Poly with many generators"""
 
     params = [1, 10, 100, 500]
 
     def setup(self, n):
-        self.xs = sympy.symbols('x:{}'.format(n))
+        self.xs = symbols('x:{}'.format(n))
         self.x = self.xs[n // 2]
-        self.px = sympy.Poly(self.x, self.xs)
+        self.px = Poly(self.x, self.xs)
 
     def time_create_poly(self, n):
-        sympy.Poly(self.x, self.xs)
+        Poly(self.x, self.xs)
 
     def time_is_linear(self, n):
         self.px.is_linear
+
+
+class TimePolyprem:
+    """Benchmark for the prem method"""
+
+    params = [(1,), (5,), (10,), (15,)]
+
+    def setup(self, size):
+        self.R, self.x, self.y, self.y1, self.y2, self.y3, self.y4, self.y5, self.y6, self.y7, self.y8, self.y9, self.y10 = ring("x, y, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10", ZZ)
+        self.y = [self.y1, self.y2, self.y3, self.y4, self.y5, self.y6, self.y7, self.y8, self.y9, self.y10]
+        self.values = {}
+        self.size = size
+
+    def generate_poly_case1(self):
+        # Linearly dense quartic inputs with quadratic GCDs.
+        n = self.size
+        D = (1 + self.x + sum(self.y[:n])) ** 2
+        self.f = D * (-2 + self.x - sum(self.y[:n])) ** 2
+        self.g = D * (2 + self.x + sum(self.y[:n])) ** 2
+
+    def time_new_prem_case1(self):
+        self.generate_poly_case1()
+        self.values['new_prem_case1'] = prem(self.f, self.g, self.x)
```

Comment:
The `generate_poly_case1` method should not be called inside the `time_` method because its time will be included in what is measured by asv. The setup method should be used to prepare the inputs to `prem` and then the `time_` method should do nothing more than the operation that is being timed (`prem`).